overlord/snapstate/snapmgr.go

```diff
@@ -1067,6 +1071,71 @@ func (m *SnapManager) localInstallCleanup() error {
 	return osutil.UnlinkManyAt(d, filenames)
 }
 
+func MockEnsuredMountsUpdated(m *SnapManager, ensured bool) (restore func()) {
+	osutil.MustBeTestBinary("ensured snap mounts can only be mocked from tests")
+	old := m.ensuredMountsUpdated
+	m.ensuredMountsUpdated = ensured
+	return func() {
+		m.ensuredMountsUpdated = old
+	}
+}
+
+func (m *SnapManager) ensureMountsUpdated() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	if m.ensuredMountsUpdated {
+		return nil
+	}
+
+	// only run after we are seeded
+	var seeded bool
+	err := m.state.Get("seeded", &seeded)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+	if !seeded {
+		return nil
+	}
+
+	allStates, err := All(m.state)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+
+	var sysd systemd.Systemd
+	if len(allStates) > 0 {
+		if snapdenv.Preseeding() {
+			sysd = systemd.NewEmulationMode(dirs.GlobalRootDir)
+		} else {
+			sysd = systemd.New(systemd.SystemMode, nil)
+		}
+	}
+
+	for _, snapSt := range allStates {
+		info, err := snapSt.CurrentInfo()
+		if err != nil {
+			return err
+		}
+		squashfsPath := dirs.StripRootDir(info.MountFile())
+		whereDir := dirs.StripRootDir(info.MountDir())
+		_, err = sysd.EnsureMountUnitFile(
```

Comment:
(super-nitpick) this could be a single line: `if _, err := sysd.EnsureMountUnitFile(...); err != nil ` (it looks like even then it's not *that* long. But feel free to ignore, it's really a minor nitpick.